tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/Refiner.scala

```diff
@@ -69,10 +69,12 @@ final class Refiner(semantics: Semantics, outputMode: OutputMode) {
     }
 
     val kind = {
-      if (classDef.kind == ClassKind.ModuleClass && !info.isModuleAccessed)
-        ClassKind.Class
-      else
+      if (classDef.kind.hasModuleAccessor && !info.isModuleAccessed) {
+        if (classDef.kind == ClassKind.ModuleClass) ClassKind.Class
+        else ClassKind.JSClass
+      } else {
         classDef.kind
+      }
```

Comment:
If you have a suggestion about *where* to put such a helper? `ClassKind`'s companion object?

Eventually (probably in the next major version), I want to entirely refactor `ClassKind`s anyway. They now conflate several axes that have become orthogonal. For example, I would reify `hasModuleAccessor` as a separate flag.